tests/tests/test-crowdloan.ts

```diff
@@ -514,3 +516,62 @@ describeDevMoonbeam("Crowdloan", (context) => {
     );
   });
 });
+
+describeDevMoonbeam("Crowdloan", (context) => {
+  let genesisAccount: KeyringPair, sudoAccount: KeyringPair;
+
+  before("Setup genesis account for substrate", async () => {
+    const keyring = new Keyring({ type: "ethereum" });
+    genesisAccount = await keyring.addFromUri(GENESIS_ACCOUNT_PRIVATE_KEY, null, "ethereum");
+    sudoAccount = await keyring.addFromUri(ALITH_PRIV_KEY, null, "ethereum");
+  });
+  it("should be able to initialize through democracy", async function () {
+    let calls = [];
+    // We are gonna put the initialization and completion in a batch_all utility call
+    calls.push(
+      context.polkadotApi.tx.crowdloanRewards.initializeRewardVec([
+        [relayChainAddress, GENESIS_ACCOUNT, 3_000_000n * GLMR],
+      ])
+    );
+
+    let initBlock = (await context.polkadotApi.query.crowdloanRewards.initRelayBlock()) as any;
+    calls.push(
+      context.polkadotApi.tx.crowdloanRewards.completeInitialization(Number(initBlock) + vesting)
+    );
+
+    // Here we build the utility call
+    const proposal = context.polkadotApi.tx.utility.batchAll(calls);
+
+    // We encode the proposal
+    let encodedProposal = (proposal as SubmittableExtrinsic)?.method.toHex() || "";
+    let encodedHash = blake2AsHex(encodedProposal);
+
+    // Submit the pre-image
+    await context.polkadotApi.tx.democracy
+      .notePreimage(encodedProposal)
+      .signAndSend(genesisAccount);
+
+    await context.createBlock();
+
+    // Propose
+    await context.polkadotApi.tx.democracy
+      .propose(encodedHash, 1000n * GLMR)
+      .signAndSend(genesisAccount);
+
+    await context.createBlock();
+    const publicPropCount = await context.polkadotApi.query.democracy.publicPropCount();
+
+    // we only use sudo to enact the proposal
+    await context.polkadotApi.tx.sudo
+      .sudoUncheckedWeight(
+        context.polkadotApi.tx.democracy.enactProposal(encodedHash, publicPropCount),
+        1
+      )
+      .signAndSend(sudoAccount);
+
+    await context.createBlock();
+
+    let isInitialized = await context.polkadotApi.query.crowdloanRewards.initialized();
+    expect(isInitialized.toHuman()).to.be.true;
```

Comment:
I wonder if we could have more thorough asserts here. For example:
Ensure the genesis account has the correct number of associated funds.
Ensure there are no unassociated accounts.
Ensure that there is the right number of associated accounts.
Ensure the first 30% was paid correctly.

Up to you about this, but it might be nice to also test that unassociated contributions are initialized correctly and that the dust mechanism works correctly.

Reply:
The dust mechanism is not yet merged, so cannot test it. Once is merged I can add a test.
The rest I can add it yes, although some of the stuff is analogous to the other tests of the file. That is the main reason why I have not made more asserts

Reply:
Okay, gotcha. I didn't notice that.

Reply:
I still added more asserts :+1: